Remove custom d2l-menu items

ryantmer opened the pull request:
Removing the custom menu item for the sorting menu, and applying `d2l-menu-item-selectable-styles` to the custom item for the filter menu.

After some discussion in https://github.com/Brightspace/d2l-menu-ui/pull/17 and offline, it was decided that the filter menu checkboxes aren't really _menu items_ per se, so they're remaining as custom items. However, their use is limited to the all courses overlay, hence keeping them there for now. If there is a use case to use them elsewhere, they should probably be broken out into their own repo.

ryantmer commented:
FYI: Updated this to leverage a change I made to `icons`, and thereby significantly simplify this (mostly @trevorjonesd2l, in case you've already started poking)

trevorjonesd2l commented:
Well I don't see any more problemos with the codeedo so I'm gonna start testing

trevorjonesd2l commented:
I'd say fix:
- [ ] [this](https://github.com/Brightspace/d2l-menu-ui/issues/18)
- [ ] The issue where hitting space on a radio button closes and reopens the dropdown

ryantmer commented:
Fixed the first and did a `d2l-menu` patch release - still haven't figured out the second problem, I'll take another look tomorrow.

ryantmer commented:
Hokay, did a bunch of poking, short answer is that the reopening of the dropdown is coming from within `d2l-dropdown-menu` (or possible `d2l-menu` itself). Either way, I've got https://github.com/Brightspace/d2l-menu-ui/pull/20 and https://github.com/Brightspace/d2l-dropdown-ui/pull/84 open to at least slightly improve things.

ryantmer commented:
Fixed weird Enter behaviour thingy in https://github.com/Brightspace/d2l-menu-ui/pull/21 (also fixed scrolling on Space when using the course tile menu) ((also fixed flat tire))

ryantmer commented:
Sigh... I've actually pushed that fix now. Good to re-check and merge whenever now :P

trevorjonesd2l commented:
Trevor test

trevorjonesd2l commented:
> There once was a PR by Ryan
> Not perfect so he kept on tryin'
> He made a commit
> And called it legit
> Trevor test and thumbs ups are flyin'

👍
